File: Runtime/OutputStratumTransitionStateCollection.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports SyncroSim.Common
Imports System.Collections.ObjectModel

Friend Class OutputStratumTransitionStateCollection
    Inherits KeyedCollection(Of EightIntegerLookupKey, OutputStratumTransitionState)

    Public Sub New()
        MyBase.New(New EightIntegerLookupKeyEqualityComparer)
    End Sub

    Protected Overrides Function GetKeyForItem(ByVal item As OutputStratumTransitionState) As EightIntegerLookupKey

        Return New EightIntegerLookupKey(
            item.StratumId,
            LookupKeyUtils.GetOutputCollectionKey(item.SecondaryStratumId),
            LookupKeyUtils.GetOutputCollectionKey(item.TertiaryStratumId),
            item.Iteration,
            item.Timestep,
            item.TransitionTypeId,
            item.StateClassId,
            item.EndStateClassId)

    End Function

End Class

File: Runtime/OutputStratumTransitionStateCollection.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using SyncroSim.Common;
using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    internal class OutputStratumTransitionStateCollection : KeyedCollection<EightIntegerLookupKey, OutputStratumTransitionState>
    {
        public OutputStratumTransitionStateCollection() : base(new EightIntegerLookupKeyEqualityComparer())
        {
        }

        protected override EightIntegerLookupKey GetKeyForItem(OutputStratumTransitionState item)
        {
            return new EightIntegerLookupKey(
                item.StratumId, 
                LookupKeyUtils.GetOutputCollectionKey(item.SecondaryStratumId), 
                LookupKeyUtils.GetOutputCollectionKey(item.TertiaryStratumId), 
                item.Iteration, item.Timestep, item.TransitionTypeId, item.StateClassId, item.EndStateClassId);
        }
    }
}
